src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -49,18 +65,24 @@ public MesosApi(String masterUrl, String user, String frameworkName)
             .withValue("master-url", ConfigValueFactory.fromAnyRef(masterUrl));
     this.clientSettings = MesosClientSettings.fromConfig(conf);
     system = ActorSystem.create("mesos-scheduler");
+    context = system.dispatcher();
     materializer = ActorMaterializer.create(system);
-
     client = connectClient().get();
+    stateMap = new ConcurrentHashMap<>();
+    adapter = new SchedulerAdapter(Scheduler.fromClient(client), materializer, context);
+    adapterSourceSink = adapter.asSourceAndSink(SpecsSnapshot.empty());
+    adapterSourceSink.t2().runForeach(this::updateState, materializer);
+    adapterQueues = adapter.asAkkaQueues(SpecsSnapshot.empty(), OverflowStrategy.backpressure());
```

Comment:
This is not how it's supposed to be used allowed. Such a call will create 2 schedulers with 2 independent mesos clients.

Reply:
ya this was more experimentation